ext/web/lib.rs

```diff
@@ -314,46 +316,29 @@ impl Resource for TextDecoderResource {
   }
 }
 
-#[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
-struct EncodeIntoResult {
-  read: usize,
-  written: usize,
-}
-
-#[op]
+#[op(v8)]
 fn op_encoding_encode_into(
-  input: String,
+  scope: &mut v8::HandleScope,
+  input: serde_v8::Value,
   buffer: &mut [u8],
-) -> EncodeIntoResult {
-  // Since `input` is already UTF-8, we can simply find the last UTF-8 code
-  // point boundary from input that fits in `buffer`, and copy the bytes up to
-  // that point.
-  let boundary = if buffer.len() >= input.len() {
-    input.len()
-  } else {
-    let mut boundary = buffer.len();
-
-    // The maximum length of a UTF-8 code point is 4 bytes.
-    for _ in 0..4 {
-      if input.is_char_boundary(boundary) {
-        break;
-      }
-      debug_assert!(boundary > 0);
-      boundary -= 1;
-    }
-
-    debug_assert!(input.is_char_boundary(boundary));
-    boundary
+  out_buf: &mut [u8],
+) {
+  let s = v8::Local::<v8::String>::try_from(input.v8_value).unwrap();
+  assert!(out_buf.len() == std::mem::size_of::<u32>() * 2);
+  // SAFETY: `out_buf` is guaranteed to be large enough to hold result.
+  let out_buf: &mut [u32] = unsafe {
+    std::slice::from_raw_parts_mut(out_buf.as_mut_ptr() as *mut u32, 2)
   };
```

Comment:
I still don't understand what guarantees alignment. Can you explain?

Note that getting the alignment wrong can lead to miscompiles, even on x86_64.

Reply:
In JS, ArrayBuffers with a length ≥ the necessary alignment of type T, are usable as that type.

If this accepts TypedArrays and turns them into `&mut [u8]`, then it's still unsound, as one may pass `new Uint8Array(5).subarray(1)`, which has a `byteOffset` that isn't aligned to a `u32` boundary.

Reply:
yikes, lemme add `&mut [u32]` support to `#[ops]` to directly deal with `Uint32Array`